src/OpenTelemetry.Api/Trace/ActivityExtensions.cs

```diff
@@ -62,8 +70,8 @@ public static Status GetStatus(this Activity activity)
         {
             Debug.Assert(activity != null, "Activity should not be null");
 
-            var statusCanonicalCode = activity.Tags.FirstOrDefault(k => k.Key == SpanAttributeConstants.StatusCodeKey).Value;
-            var statusDescription = activity.Tags.FirstOrDefault(d => d.Key == SpanAttributeConstants.StatusDescriptionKey).Value;
+            var statusCanonicalCode = activity.GetTagValue(SpanAttributeConstants.StatusCodeKey) as string;
+            var statusDescription = activity.GetTagValue(SpanAttributeConstants.StatusDescriptionKey) as string;
```

Comment:
Minor: this is a potential to do one loop for two keys. Not blocking this PR though.

Reply:
I commented above about this.